UsageEnvironment/Log.cpp

```diff
@@ -77,20 +86,127 @@ FILE* _Log::AcquireFile() {
   } else {
     FILE* logFileHandle = fopen(logFilePath, "a");
     if( logFileHandle == NULL ) {
-      strcpy(logFilePath, "/tmp/live555.log");
+#ifdef LOG_FILE_TMP_DIR
+      strcpy(logFilePath, LOG_FILE_TMP_DIR "/live555.log");
       logFileHandle = fopen(logFilePath, "a");
+#endif
     }
 
     return logFileHandle;
   }
 }
 
+void _Log::RemoveOldBackup() {
+#ifdef LOG_FILE_BACKUP_DIR
+  DIR* backupDir;
+  struct dirent* dirInfo;
+
+  // Look for old backup log file and remove it.
+  backupDir = opendir(LOG_FILE_BACKUP_DIR);
+  if( backupDir ) {
+    while( (dirInfo = readdir(backupDir)) != NULL ) {
+      if( strncmp(dirInfo->d_name, "live555", 7) == 0 ) {
+        char rmBuf[PATH_MAX];
+
+        // Attempt to remove using system call (unlink() and remove() complain /mnt/config is Read-Only)
+        fprintf(stdout, "Attempting to remove old backup log.\n");
+        snprintf(rmBuf, sizeof(rmBuf), "rm " LOG_FILE_BACKUP_DIR "/%s", dirInfo->d_name);
+        system(rmBuf);
+        break;
+      }
+    }
+  }
+  closedir(backupDir);
+#endif
+}
```

Comment:
If there's only every going to be one log file in Flash memory at any given time, why not use a constant name that doesn't have the date/time appended? Then you don't need to search for it, you just delete it.  (Appending a date/time stamp to a filename is generally only done when you anticipate having multiple versions of a file in the same directory.)

Also, ideally the filename should be specified by the client. We discussed adding a trigger to make the dump happen. The trigger can pass in the full filepath to where the dump file should be generated. This way the client, in this case `rtspd`, can then manage where to put the file, how many copies to keep, etc.

Reply:
I feel like for the backup location it makes more sense to have it as a build macro instead of a parameter passed into BackupLog() so it's a consistent location across all other functions. Then I feel like it might be a better idea to just have the filename of the backup be passed into BackupLog() since having multiple backup locations doesn't seem to be a likely use-case and it feels less abstracted to have rtspd handle different locations especially since this will affect removing backups as well, I prefer to just have the logger handle the backing up to one location and maybe different errors could have different file names for the backups.